ext/standard/var.c

```diff
@@ -145,14 +146,22 @@ PHPAPI void php_var_dump(zval *struc, int level) /* {{{ */
 			PUTS("}\n");
 			break;
 		case IS_OBJECT:
+			class_name = Z_OBJ_HANDLER_P(struc, get_class_name)(Z_OBJ_P(struc));
+
+			zend_class_entry *ce = Z_OBJCE_P(struc);
+			if (ce->ce_flags & ZEND_ACC_ENUM) {
+				zval *case_name_zval = zend_enum_fetch_case_name(Z_OBJ_P(struc));
+				php_printf("enum(%s::%s)\n", ZSTR_VAL(class_name), Z_STRVAL_P(case_name_zval));
```

Comment:
nit on trivial implementation detail: Sorry that I missed this during earlier reviews, this is easy to miss and I assume it was just overlooked by reviewers.

var_dump()'s output prepends `&` to any zvals that are PHP references - PHP `#define COMMON (is_ref ? "&" : "")`. Would it make sense to do the same thing for enums as well? (for non-enums, var_dump dumps the object id of objects, but also dumps whether the object is a reference, and knowing that it is a reference makes debugging easier (e.g. in phpt test cases))

```suggestion
				php_printf("%senum(%s::%s)\n", COMMON, ZSTR_VAL(class_name), Z_STRVAL_P(case_name_zval));
```

```php
$arr = [Suit::Hearts];
$arr[1] = &$arr[0];
var_dump($arr);
```

Reply:
I'm not sure if that's a bug. You shouldn't be allowed to reference enums.

Reply:
Oh sorry, of course you are. You're not allowed to use their properties as references. Yeah, this makes sense then I suppose.

Reply:
Hm, apparently this doesn't seem to work for objects in general. https://3v4l.org/S2fZq

Reply:
Anyway, I added a test for your case and that seems to work.